vcd/resource_vcd_org_vdc_access_control.go

```diff
@@ -0,0 +1,228 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"strings"
+)
+
+func resourceVcdOrgVdcAccessControl() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdVdcAccessControlCreateUpdate,
+		ReadContext:   resourceVcdVdcAccessControlRead,
+		UpdateContext: resourceVcdVdcAccessControlCreateUpdate,
+		DeleteContext: resourceVcdVdcAccessControlDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdVdcAccessControlImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"shared_with_everyone": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Whether the VDC is shared with everyone",
+			},
+			"everyone_access_level": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{types.ControlAccessReadOnly}, true),
+				Description:  "Access level when the VDC is shared with everyone (only ReadOnly is available). Required when shared_with_everyone is set",
```

Comment:
You could put the following schema validation attribute so that it is validated at `terraform plan` instead of runtime / api calls.

`RequiredWith: []string{"shared_with_everyone"},`

Reply:
I don't think so, since `shared_with_everyone` is required, and depending if it is true or false, I need to check either `everyone_access_level` or `shared_with` at runtime (to avoid inconsistencies).

Reply:
Ok. This is good - these are the limitation of Terraform, but when we can use them - it is way more convenient:

<img width="720" alt="image" src="https://user-images.githubusercontent.com/15804230/171137259-de71f03d-0fe6-404b-84a1-c68cc3920012.png">